spirv_glsl.cpp

```diff
@@ -3258,6 +3306,78 @@ void CompilerGLSL::emit_resources()
 	declare_undefined_values();
 }
 
+void CompilerGLSL::emit_extension_workarounds()
+{
+	//GL_KHR_shader_subgroup_vote
+	if (!options.vulkan_semantics && declaredKHR_shader_subgroup_vote)
+	{
+		statement("#ifdef GL_ARB_shader_group_vote");
+		statement("bool subgroupAll(in bool v) { return allInvocationsARB(v); }");
+		statement("bool subgroupAny(in bool v) { return anyInvocationARB(v); }");
+		statement("#define SUBGROUP_ALL_EQUAL_WORKAROUND(type) bool subgroupAllEqual(type value) { return "
+		          "allInvocationsEqualARB(equal(value)); }");
+		statement("SUBGROUP_ALL_EQUAL_WORKAROUND(int)");
+		statement("SUBGROUP_ALL_EQUAL_WORKAROUND(ivec2)");
```

Comment:
Can have an array of the type names and just emit statement in a loop to keep the LOC down.